### PR TITLE
fix(ui): TE-2130 add checkbox to slack edit and view for subscription group spec

### PR DESCRIPTION
**Pull request by alcatraz627**
#### Issue(s)

[TE-2130](https://startree.atlassian.net/browse/TE-2130)
[UX Mocks](https://www.figma.com/file/9pj4WGZBSNweOeeRKGeSnF/Subscription-groups-%2F-New-feature-%2F-Notification-when-anomaly-is-over?type=design&node-id=612-31389&mode=design&t=iJ9ilvkcHaFldUWB-0)

#### Description

- Add a checkbox for `Notify when the anomaly period ends` for slack spec in subscription groups
- Show the corresponding value as readonly in spec view

#### Screenshots

Screenshot 2024-03-12 at 19 22 58

Screenshot 2024-03-12 at 19 35 50

### CODE REVIEW

**Comment by jayeshchoudhary** [comment]
it makes sense to show checkbox when taking input. but showing checkbox as a value doesn't seem right.

**Comment by alcatraz627** [comment]
@jayeshchoudhary felt so too, but using checkbox in one place and switch in the other felt contradictory 

Let me take this up with Edwin

Screenshot 2024-03-12 at 19 22 52

File: thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/recipient-details/groups-editor/slack/slack.component.tsx [hunk 1]
```diff
@@ -146,6 +159,40 @@ export const Slack: FunctionComponent<SlackProps> = ({
...
+                        control={control}
+                        name="notifyResolvedAnomalies"
+                        render={({ field: { name, value, onChange } }) => (
+                            <InputSection
+                                inputComponent={
+                                    <FormControlLabel
+                                        // eslint-disable-next-line max-len
+                                        // `control` needs to be passed on differently for checkbox
+                                        // eslint-disable-next-line max-len
+                                        // https://stackoverflow.com/questions/62291962/react-hook-forms-material-ui-checkboxes
```
Comment (jayeshchoudhary) [inline thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/recipient-details/groups-editor/slack/slack.component.tsx:173]: imo this is obvious and documented in the mui docs as well
can be removed.

**Comment by jayeshchoudhary** [comment]
> @jayeshchoudhary felt so too, but using checkbox in one place and switch in the other felt contradictory
> 
> Let me take this up with Edwin
> 
> Screenshot 2024-03-12 at 19 22 52

true false / YES NO??

File: thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/recipient-details/groups-editor/slack/slack.component.tsx (hunk 1 above)
Comment (alcatraz627) [reply thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/recipient-details/groups-editor/slack/slack.component.tsx:173]: Ah, this comment was for my previous code at this (which did not work for the initial state)

Removing.

**Review by jayeshchoudhary — APPROVED** [review]

**Pull request merged**